crates/levm/src/vm.rs

```diff
@@ -137,33 +155,27 @@ fn address_to_word(address: Address) -> U256 {
 // tuple of environmental data.
 
 impl VM {
-    // TODO: block and transaction, not this
-    pub fn new(bytecode: Bytes, address: Address, balance: U256) -> Self {
-        let initial_account = Account::new(balance, bytecode.clone());
+    pub fn new(tx_env: TxEnv, block_env: BlockEnv, db: Db) -> Self {
+        let bytecode = match tx_env.transact_to {
+            TransactTo::Call(addr) => db.get_account_bytecode(&addr),
+            TransactTo::Create => {
+                todo!()
+            }
+        };
 
         let initial_call_frame = CallFrame::new_from_bytecode(bytecode);
```

Comment:
Here I would use the data that we bring from the `TxEnv` to create a new `CallFrame` in the same way that we do on `generic_call`, with the method `CallFrame::new`.
```suggestion
        let initial_call_frame = CallFrame::new(
            gas,
            msg_sender,
            to,
            code_address,
            delegate,
            callee_bytecode,
            value,
            calldata,
            is_static,
        );
```

Reply:
I created it with some of the `tx_env` fields, but I left some as default for now.